src/core/meshcop/dataset_manager.cpp

```diff
@@ -446,7 +471,8 @@ ThreadError DatasetManager::Set(Coap::Header &aHeader, Message &aMessage, const
         mLocal.Set(mNetif.GetActiveDataset().GetNetwork());
     }
 
-    if (!isUpdateAffectConnectivity)
+    if (!isUpdateAffectConnectivity ||
+        (isUpdateAffectConnectivity && isUpdateIdenticalValue))
```

Comment:
The second instance of `isUpdateAffectConnectivity` is redundant.  I believe, the following is equivalent:

```
if (!isUpdateAffectConnectivity || isUpdateIdenticalValue)
```